tests/test_structs.py

```diff
@@ -136,3 +136,61 @@ def test_frozen_trial_repr():
                         intermediate_values={})
 
     assert trial == eval(repr(trial))
+
+
+def test_study_summary_study_id():
+    # type: () -> None
+
+    study = optuna.create_study()
+    summaries = study._storage.get_all_study_summaries()
+    assert len(summaries) == 1
+
+    summary = summaries[0]
+
+    assert summary.study_id == summary._study_id
+    with pytest.warns(DeprecationWarning):
+        summary.study_id
+
+
+def test_study_summary_eq_ne_lt():
+    # type: () -> None
+
+    storage = optuna.storages.RDBStorage('sqlite:///:memory:')
+
+    optuna.create_study(storage=storage)
+    study = optuna.create_study(storage=storage)
+
+    summaries = study._storage.get_all_study_summaries()
+    assert len(summaries) == 2
+
+    assert summaries[0] == copy.deepcopy(summaries[0])
+    assert summaries[0] != summaries[1]
+
+
+# TODO(Yanase): Remove version check after Python 2.7 is retired.
+@pytest.mark.skipif('sys.version_info < (3, 5)')
```

Comment:
Thank you for your comment. I added the reason in this commit (https://github.com/optuna/optuna/pull/718/commits/6623115e10629bd9dd987fdff6233414343b3861). As far as I know, `NotImplemented` raises `TypeError` in Python 3, but I'm not if it is defined in Python specification. If you find it inappropriate, please let me know.

Reply:
Ah, I see. Just leaving some related notes https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons. To be picky, we should probably try to align the behaviors between the Python versions but considering the fact that we're dropping Python 2 soon and that it's not very likely to compare a `StudySummary` with a different type this may be fine.

Reply:
> The ordering comparison operators (<, <=, >=, >) raise a TypeError exception when the operands don’t have a meaningful natural ordering. 
(from https://docs.python.org/3.0/whatsnew/3.0.html#ordering-comparisons)

Thank you for the reference. Now, I understand we can expect `TypeError` in Python 3.

And, I agree with your opinion about the inconsistent behavior between Python 2 and 3. I'd like to keep the current implementation for simplicity.